pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -540,13 +526,27 @@ public RepositoryData resolveAndRepackage() {
                                                         extensionRtArtifact.getExtension(),
                                                         null),
                                                 "compile")),
-                                bomConstraints, // version constraints from the BOM
+                                new ArrayList<>(bomConstraints), // version constraints from the BOM
                                 Collections.emptyList(), // extra maven repos, ignore this
                                 "test",
                                 "provided" // dependency scopes that should be ignored
                         );
                     }
                 }
+                bannedDirs.entrySet()
+                        .stream()
+                        .filter(en -> Files.exists(en.getValue()))
+                        .peek(en -> bannedReport.append("\n " + en.getKey() + " pulled by " + extensionRtArtifact))
+                        .forEach(en -> {
+                            try {
+                                org.apache.commons.io.FileUtils.deleteDirectory(en.getValue().toFile());
```

Comment:
First, because if the artifact is left there, then it is reported to be pulled by each subsequent resolution (which would not necessarily be correct). Second, if some banned artifacts are found the build fails, so it does not matter that the local repo is incorrect.